src/plugins/intel_cpu/src/perf_count.cpp

```diff
@@ -0,0 +1,464 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#ifdef CPU_DEBUG_CAPS
+
+#include "perf_count.h"
+#include "exec_network.h"
+#include <fstream>
+#include <iomanip>
+
+namespace ov {
+namespace intel_cpu {
+
+static double perfAvg(const uint64_t sum, const uint64_t num);
+static double perfAvg(const double sum, const uint64_t num);
+static double perfPercent(const uint64_t val, const uint64_t total);
+static double perfDeviationPercent(const std::vector<double>& values, const double sum);
+static void perfShapeToStr(const VectorDims& shape, std::string& str);
+static void perfDumpNodes(const std::string& path, const std::vector<MKLDNNNodePtr>& nodes, const std::vector<std::string>& modelInputShapes);
+
+void PerfCount::finish_itr(const PerfKey itrKey, PerfData::PerfNodeShape&& itrNodeShape) {
+    finish_itr();
+
+    assert(itrKey <= _perfData.size());
+    if (itrKey == _perfData.size())
+        _perfData.emplace_back();
+
+    auto& perfData = _perfData[itrKey];
+    for (uint8_t i = 0; i < NumberOfCounters; i++) {
+        if (_itrDuration[i] != std::chrono::high_resolution_clock::duration::zero()) {
+            perfData.counters[i].duration += std::chrono::duration_cast<std::chrono::microseconds>(_itrDuration[i]).count();
+            perfData.counters[i].num++;
+        }
+    }
+    perfData.nodeShapesSet.insert(itrNodeShape);
+}
+
+PerfHelper::PerfHelper(const std::shared_ptr<MKLDNNNode>& node, const PerfKey itrKey) :
+        _node(node), _itrKey(itrKey) {
+    _node->PerfCounter().start_itr();
+}
+
+PerfHelper::~PerfHelper() {
+    if (_itrKey != std::numeric_limits<PerfKey>::max()) {
+        std::vector<VectorDims> in, out;
+        in.reserve(_node->getParentEdges().size());
```

Comment:
this code block takes about 80+% of total counter growth, let's move it after the timer.